pkg/types/vsphere/validation/machinepool.go

```diff
@@ -35,5 +36,28 @@ func ValidateMachinePool(p *vsphere.MachinePool, fldPath *field.Path) field.Erro
 	} else if p.NumCoresPerSocket > defaultNumCPUs {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), p.NumCoresPerSocket, "cores per socket must be less than number of CPUs which is by default 4"))
 	}
+
+	if len(p.Zones) > 0 {
+		if len(platform.DeploymentZones) == 0 {
+			return append(allErrs, field.Required(fldPath.Child("zones"), "deploymentZones must be defined if zones are defined"))
+		}
+		for _, zone := range p.Zones {
+			err := validate.ClusterName1035(zone)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("zones"), p.Zones, err.Error()))
+			}
+			zoneDefined := false
+			for _, deploymentZone := range platform.DeploymentZones {
+				if deploymentZone.Name == zone {
+					zoneDefined = true
+				}
+			}
+			if zoneDefined == false {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("zones"), zone, "zone not defined in deploymentZones"))
+			}
+		}
+	} else if len(platform.DeploymentZones) > 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("zones"), "zones must be defined if deploymentZones are defined"))
+	}
```

Comment:
Would it be a better UX, if `deploymentZones` were defined but no zones, the machinesets were distributed over the defined `deploymentZones`? I imagine that would be the common use case, and would simplify the config. Then `zones` would only need to be defined in more specific configurations.

Reply:
`DeploymentSuitable` might add a wrinkle here. We would need to consider how to handle the control plane in an edge case where, say, all provided zones were `ControlPlane == NotAllowed`